src/steamship/data/chat.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from steamship import File, MimeTypes, SteamshipError
+from steamship.base.client import Client
+from steamship.base.request import IdentifierRequest
+from steamship.base.tasks import Task
+from steamship.data import TagKind
+from steamship.data.block import Block
+from steamship.data.file import ListFileRequest, ListFileResponse
+from steamship.data.tags import Tag
+from steamship.data.tags.tag_constants import DocTag, RoleTag
+
+if TYPE_CHECKING:
+    from steamship.data.operations.generator import GenerateResponse
+
+
+class ListChatResponse:  # not a pydantic model
+    chats: List[Chat]
+
+    @staticmethod
+    def from_list_file_response(list_file_response: ListFileResponse) -> ListChatResponse:
+        result = ListChatResponse()
+        result.chats = [Chat(file) for file in list_file_response.files]
+        return result
+
+
+CHAT_GENERATOR_INSTANCE_HANDLE = "chat-generator-instance-handle"
+
+
+class Chat:
+    """A Chat is a subclass of a File ideal for ongoing interactions between a user and a virtual assistant."""
+
+    generator_instance_handle: str
+    file: File
```

Comment:
two thoughts here:
1) what if we take some sort of `GeneratorMap` instead so that we could build a subclass of `Chat` that is `MultiModalChat` that could dynamically select differing generators, etc.?  Short of that, I think we should take a list of handles.


GeneratorMap could maybe look something like this rough sketch:
```python
class BaseGenerator(BaseModel):
   handle: str
   mime_types: List[str]
   

class ImageGenerator(BaseGenerator):
   mime_types: ["png", ...]


class GeneratorMap():
  def for_mime_type() -> BaseGenerator:
       pass

  def for_handle() -> BaseGenerator:
      pass
```

2) I think the instinct to wrap `File` is a good one, but what if we wrap it as `ChatHistory`?  And have `Chat` use `ChatHistory` for memory? It would allow differing approaches towards storage in the future (tailored as needed).